File: sgv/sgv/report/item_query/item_query.js
// Copyright (c) 2022, Richard Kwa and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Item Query"] = {
	"filters": get_filters(),
	"formatter":function(value,row,column,data,default_formatter)
		{ if(data && column.fieldname=="item_code")
			value = data.item_code
			column.link_onclick = "test"
			return value;
		}
	
}
function(frm){
	$("input[data-fieldname=item_code]").mouseover(function(){
	console.log(“Hello”);

function get_filters() {
	let filters = [
		{
			"fieldname":"company",
			"label": __("Company"),
			"fieldtype":"Link",
			"options": "Company",
			"default":frappe.defaults.get_user_default("Company"),
			"reqd":1
		},
		{
			"fieldname":"item_code",
			"label": __("Item Code"),
			"fieldtype":"Data",
			"on_change":function(query_reports){
				var item_code=query_reports.get_values().item_code;
				frappe.msgprint("onchange")
			}
		},
		{
			"fieldname":"item_group",
			"label": __("Item Group"),
			"fieldtype":"Link",
			"options": "Item Group",
		},
		{
			"fieldname":"brand",
			"label": __("Brand"),
			"fieldtype":"Link",
			"options": "Brand",
			"default": "CRUCIAL"
		}
	]
	return filters;
}

File: sgv/sgv/report/item_query/item_query.py
# Copyright (c) 2022, Richard Kwa and contributors
# For license information, please see license.txt

import frappe
from frappe import _

def execute(filters=None):
	return get_columns(), get_data(filters)

def get_data(filters):
	conditions = ""

	# if filters.get('item_code'):
	#	conditions += " item_code LIKE '{filters.get('item_code')}'"
	# if filters.get('item_name'):
	#	conditions += " item_name LIKE '{filters.get('item_name')}'"
	#	return conditions
	get_data = frappe.db.sql(f"""Select item_code, item_name, item_group, brand from tabItem 
		WHERE item_code LIKE '{filters.get('item_code')}' OR item_group ='{filters.get('item_group')}' 
		OR brand='{filters.get('brand')}'""",as_dict=1)
	# get_data = frappe.db.sql(f"""Select item_code, item_name, item_group, brand from tabItem;""",filters,as_dict=True)
	return get_data

def get_columns():
	return [
		{"label":_("Item"),"fieldname":"item_code","fieldtype":"Data","width":150},
		{"label":_("Item Name"),"fieldname":"item_name","fieldtype":"Data","width":400},
		{"label":_("Item Group"),"fieldname":"item_group","fieldtype":"Data","width":150},
		{"label":_("Brand"),"fieldname":"brand","fieldtype":"Data","width":120}
	]